Switch to abi3 builds

weiji14 opened the pull request:
Create a single abi3 package per platform for non-free-threaded builds.

References:
- https://labs.quansight.org/blog/conda-abi3
- https://rattler.build/latest/reference/recipe_file/#version-independent-abi3-packages
- https://conda-forge.zulipchat.com/#narrow/channel/457337-general/topic/Python.20ABI3.20package.20with.20v1.20format

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

Closes #4 (Python 3.14 migration) and closes #3.

<!--
Please add any other relevant info below:
-->

weiji14 committed:
Switch to abi3 builds

Xref https://conda-forge.zulipchat.com/#narrow/channel/457337-general/topic/Python.20ABI3.20package.20with.20v1.20format/with/531737289. Script tests based on https://github.com/conda-forge/hf-xet-feedstock/pull/10

weiji14 committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.41.0
- rattler-build-conda-compat 1.4.6

weiji14 committed:
Keep freethreading builds with no abi3

weiji14 committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

weiji14 committed:
Make tests pass on is_freethreading builds

Only run abi3audit on is_abi3 (i.e. no is_freethreading). Need to add an `if: not is_abi3` condition to satisfy rattler-build, see similar case at https://github.com/conda-forge/torchvision-feedstock/pull/111/commits/a85e793bcea3171e65401bafb7d881ad397395c6. Also need to make python-freethreading a runtime dependency to ensure python=3.13t is used during the import tests.

weiji14 committed:
Bump build number

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

weiji14 commented:
Will probably apply these changes in the next release. Just keeping this in draft for reference.

regro-cf-autotick-bot committed:
Rebuild for python 3.14

weiji14 committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.01.17.11.50

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6
